persistence-zookeeper/src/main/scala/com/mesosphere/usi/storage/zookeeper/PodRecordRepository.scala

```diff
@@ -0,0 +1,48 @@
+package com.mesosphere.usi.storage.zookeeper
+
+import akka.util.ByteString
+import com.mesosphere.usi.core.models.{PodId, PodRecord}
+import com.mesosphere.usi.repository.{PodRecordRepository => PodRecordRepositoryInterface}
+import com.mesosphere.usi.storage.zookeeper.PersistenceStore.Node
+import org.apache.zookeeper.KeeperException.NoNodeException
+
+import scala.async.Async.{async, await}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class PodRecordRepository(val store: PersistenceStore) extends PodRecordRepositoryInterface {
+
+  override def create(record: PodRecord): Future[PodId] = async {
+    val data = record.toString
+    val node = Node(record.podId.value, ByteString(data))
```

Comment:
Keep in mind - you want to store more entity types than just `PodRecord`s. You need some prefix for all _PodRecords_ like `/records` so that resulting path looks like `/records/eng/sleep`

Reply:
Also: you're using `podId.value` as a node path - does this mean, there is always only 1 `PodRecord` for a given `podId`? What about pod versions? (podId stays the same, but RunSpec changes)

Reply:
We have to talk about this. I don't think `PodRecord`s will have paths as their id. We can have versioning.